src/Neo/SmartContract/ApplicationEngine.cs

```diff
@@ -319,7 +319,11 @@ private ExecutionContext CallContractInternal(ContractState contract, ContractMe
             state.CallingContext = currentContext;
 
             for (int i = args.Count - 1; i >= 0; i--)
+            {
+                if (!CheckItemType(args[i], method.Parameters[i].Type))
+                    throw new InvalidOperationException($"The type of the argument `{args[i]}` does not match the formal parameter.");
```

Comment:
> It may be too harsh for mainnet contracts (even if enabled with hardfrok), this change may break some of them and even prevent them from update. What do you think about using the same approach as it was done for SC Events parameters check ([#2810 (comment)](https://github.com/neo-project/neo/pull/2810#issuecomment-1254222026)): in next release we firstly enable a warning log in case of parameters mismatch, and then in the subsequent release we convert the log to an exception.

We have discussed this in the last meeting, check will only happen for newly deployed or updated contracts, this pr will be updated.